core/utils/profiler.hpp

```diff
@@ -5,30 +5,35 @@
 #include <chrono>
 
 class TicToc {
-  std::string name_;
+  std::string_view name_;
   const kagome::log::Logger &log_;
   std::chrono::time_point<std::chrono::high_resolution_clock> t_;
 
  public:
-  TicToc(const std::string &name, const kagome::log::Logger &log)
+  TicToc(std::string_view name, const kagome::log::Logger &log)
       : name_(name), log_(log) {
     t_ = std::chrono::high_resolution_clock::now();
   }
 
-  void toc(int line = -1) {
+  template <int32_t kLine = -1>
+  inline void toc(int line = -1) {
```

Comment:
Why? It made to do `tick.toc(__LINE__);`

Reply:
Maybe template compile time argument is expected to work faster than usual argument (`.toc<__LINE__>()`)